src/clamp.test.ts

```diff
@@ -0,0 +1,11 @@
+import assert from "assert";
+import {clamp} from "./clamp";
+
+describe("clamp", () => {
+  it("basics", () => {
+    assert.strictEqual(clamp(undefined,1,10,0),1);
```

Comment:
I'd like to also assert that we clamp an `undefined` correctly when the default is within or above the bounds.

- `assert.strictEqual(clamp(undefined,1,10,5),5);`
- `assert.strictEqual(clamp(undefined,1,10,15),15);`

That's a bit of an opinion. Technically the existing tests hit all the code paths, but writing tests that more thoroughly enumerate the possible inputs results in tests that are more likely to catch a new bug when the subject is refactored.